From 759b921e3b5b2422787c9381140c27d065c6bd48 Mon Sep 17 00:00:00 2001
From: Clemens Lang <cal@macports.org>
Date: Wed, 19 Jul 2023 10:41:37 +0200
Subject: [PATCH] Avoid clang 15 crash by using correct array size

Clang 15 crashes when compiling this code because the size of the array
does not match the initializer. This is fixed in clang 16, but

  Apple clang version 14.0.3 (clang-1403.0.22.14.1)

which is currently shipped in Ventura is still affected by this.

See https://github.com/llvm/llvm-project/issues/56016 for the upstream
report of this problem.

Signed-off-by: Clemens Lang <cal@macports.org>
Upstream-Status: Submitted [https://github.com/textmate/textmate/pull/1463]
---
 Frameworks/Find/src/Find.mm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Frameworks/Find/src/Find.mm b/Frameworks/Find/src/Find.mm
index 39cbbe43..e2104fe4 100644
--- ./Frameworks/Find/src/Find.mm
+++ ./Frameworks/Find/src/Find.mm
@@ -1017,7 +1017,7 @@ static NSButton* OakCreateHistoryButton (NSString* toolTip)
 
 - (void)didFind:(NSUInteger)aNumber occurrencesOf:(NSString*)aFindString atPosition:(text::pos_t const&)aPosition wrapped:(BOOL)didWrap
 {
-	static std::string const formatStrings[4][3] = {
+	static std::string const formatStrings[2][3] = {
 		{ "No more occurrences of “${found}”.", "Found “${found}”${line:+ at line ${line}, column ${column}}.",               "${count} occurrences of “${found}”." },
 		{ "No more matches for “${found}”.",    "Found one match for “${found}”${line:+ at line ${line}, column ${column}}.", "${count} matches for “${found}”."    },
 	};
-- 
2.41.0

